tests/integration/components/welcome-message-test.js

```diff
@@ -0,0 +1,73 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import setupMirage from '../../helpers/setup-mirage';
+
+module('Integration | Component | welcome-message', function(hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  test('it renders', async function(assert) {
+    assert.expect(2);
+    const user = this.server.create('user', {});
+
+    this.session = this.owner.lookup('service:session');
+    this.session.loginUser(user);
+
+    await render(hbs`{{welcome-message}}`);
+
+    assert.equal(
+      this.element.textContent.trim(),
+      'Welcome to crates.io! Visit account settings to verify your email address and create an API token!',
+      'should show right message',
+    );
+    assert.equal(this.element.querySelector('#welcome-message').className, 'show info', 'should have right class');
+  });
+
+  test('it show reminder about email only if user has tokens', async function(assert) {
+    assert.expect(2);
+    const user = this.server.create('user', 'withTokens');
+
+    this.session = this.owner.lookup('service:session');
+    this.session.loginUser(user);
+
+    await render(hbs`{{welcome-message}}`);
+
+    assert.equal(
```

Comment:
these checks should use qunit-dom assertions

Reply:
ok. thanks. Will fix.

Reply:
Done.